arrows/core/transform_detected_object_set.cxx

```diff
@@ -0,0 +1,279 @@
+{
+  // Get base config from base class
+  vital::config_block_sptr config = vital::algorithm::get_configuration();
+
+  config->set_value( "src_camera_krtd_file_name", src_camera_krtd_file_name,
+                     "Source camera KRTD file name path" );
+
+  config->set_value( "dest_camera_krtd_file_name", dest_camera_krtd_file_name,
+                     "Destination camera KRTD file name path" );
+
+  return config;
+}
+
+
+// ------------------------------------------------------------------
+void
+transform_detected_object_set::
+set_configuration( vital::config_block_sptr config_in )
+{
+  vital::config_block_sptr config = this->get_configuration();
+
+  config->merge_config( config_in );
+  this->src_camera_krtd_file_name = config->get_value< std::string > ( "src_camera_krtd_file_name" );
+  this->dest_camera_krtd_file_name = config->get_value< std::string > ( "dest_camera_krtd_file_name" );
+
+  this->src_camera = kwiver::vital::read_krtd_file( this->src_camera_krtd_file_name );
+  this->dest_camera = kwiver::vital::read_krtd_file( this->dest_camera_krtd_file_name );
+}
+
+
+// ------------------------------------------------------------------
+bool
+transform_detected_object_set::
+check_configuration( vital::config_block_sptr config ) const
+{
+  kwiver::vital::config_difference cd( this->get_configuration(), config );
+  const auto key_list = cd.extra_keys();
+
+  if ( ! key_list.empty() )
+  {
+    LOG_WARN( logger(), "Additional parameters found in config block that are not required or desired: "
+              << kwiver::vital::join( key_list, ", " ) );
+  }
+
+  return true;
+}
+
+
+// ------------------------------------------------------------------
+Eigen::Vector3d
+transform_detected_object_set::
+backproject_to_height( const kwiver::vital::camera_perspective_sptr camera,
+		       const Eigen::Vector2d img_pt ) const
+{
+  // Back an image point to a specified world height
+  double height = 0.0;
+
+  Eigen::Vector3d img_pt_3;
+  img_pt_3 << img_pt, 1.0;
+
+  Eigen::Vector3d npt = camera->intrinsics()->as_matrix().colPivHouseholderQr().solve( img_pt_3 );
+  Eigen::Matrix3d M = camera->rotation().matrix();
+  Eigen::Vector3d t = camera->translation();
+
+  M(0,2) = height * M(0,2) + t(0);
+  M(1,2) = height * M(1,2) + t(1);
+  M(2,2) = height * M(2,2) + t(2);
+
+  Eigen::Vector3d wpt = M.colPivHouseholderQr().solve( npt );
+  auto output = Eigen::Vector3d(wpt(0) / wpt(2), wpt(1) / wpt(2), height);
+
+  return output;
+}
+
+// ------------------------------------------------------------------
+Eigen::Vector3d
+transform_detected_object_set::
+backproject_to_plane( const kwiver::vital::camera_perspective_sptr camera,
+		      const Eigen::Vector2d img_pt,
+		      const Eigen::Vector4d plane ) const
+{
+  // Back an image point to a specified world plane
+  Eigen::Vector3d img_pt_3;
+  img_pt_3 << img_pt, 1.0;
+
+  Eigen::Vector3d npt = camera->intrinsics()->as_matrix().colPivHouseholderQr().solve( img_pt_3 );
```

Comment:
This will not handle radial distortion, but the `camera_intrinsics` class already has a function that does what you need.

    vector_2d npt = camera->intrinsics()->unmap(img_pt);

Then append the 1.0 to this 2D vector to make it homogeneous (3D)